src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java

```diff
@@ -993,39 +998,37 @@ public CompactionTasks getUserDefinedTasks(Collection<SSTableReader> sstables, i
         }
     }
 
+    @Override
     public int getEstimatedRemainingTasks()
     {
-        maybeReloadDiskBoundaries();
-        int tasks = 0;
-        readLock.lock();
-        try
-        {
-            for (AbstractCompactionStrategy strategy : getAllStrategies())
-                tasks += strategy.getEstimatedRemainingTasks();
-        }
-        finally
-        {
-            readLock.unlock();
-        }
-        return tasks;
+        return getStrategies(false).stream()
+                                   .flatMap(list -> list.stream())
+                                   .map(CompactionStrategy::getEstimatedRemainingTasks)
```

Comment:
Nit: This could be a little more efficient as `.mapToInt(...).sum()`.

Reply:
Fixed in c0c1469b71e5a29fca98d7d5dc4c210e1ddc2812.